Freedom_of_Computing

tpapak01 opened the pull request:
The only changes outside of our FOC module are in the common "DemoCommunity" class, we just added some functions for message sending and receiving, without affecting any functionality
Feel free to check our module out

MattSkala commented:
I see there are some compiler warnings, as well as minor ktlint issues. You should see them after running  `./gradlew check --no-build-cache`. Can you please fix those?

I would recommend you to treat compiler and lint warnings as errors by settings `android.kotlinOptions.allWarningsAsErrors = true` and `ktlint.ignoreFailures = false` in your `build.gradle`, [in the same way as I did for the common module](https://github.com/Tribler/trustchain-superapp/commit/5915ca2c8195754aa881b4a3efae5a7db10db1e0).

Also please rebase to the latest `dao` branch, I fixed CI configuration so now it should run a check for merge requests automatically.

tpapak01 commented:
We fixed the formatting issues.
We also updated our build.gradle and fixed most compiler warnings, apart from the following:

> Task :FOC:compileReleaseKotlin
w: C:\...\FOC\MainActivityFOC.kt: (193, 25): 'getExternalStorageDirectory(): File!' is deprecated. Deprecated in Java
w: C:\...\FOC\MainActivityFOC.kt: (292, 29): 'getExternalStorageDirectory(): File!' is deprecated. Deprecated in Java
w: C:\...\FOC\MainActivityFOC.kt: (308, 72): Unchecked cast: Class<*>! to Class<Any>

The reason is we don't yet know alternatives to using those deprecated functions, and in the last case, the cast needs to be unchecked (we think) because of what we are implementing.

MattSkala commented:
That's OK, you can suppress warnings on a statement level using `@Supress` annotation. Android Studio should tell you the exact syntax if you inspect available options for the warning. In your case, I believe it would be `@Suppress("DEPRECATION")` and `@Suppress("UNCHECKED_CAST")`.
